addons/account/tests/test_account_move_entry.py

```diff
@@ -494,6 +494,148 @@ def test_journal_override_sequence_regex(self):
+        move.action_post()  # now we are fine
+
+        # move._compute_name()
+        self.assertEqual(move.name, 'FA20213335')
+        self.assertEqual(move.sequence_prefix, 'FA2021')
+        self.assertEqual(move.sequence_number, 3335)
+
+    def test_journal_override_sequence_no_reset2(self):
+        """
+        The default regex matches many cases but fails to update the fields
+        """
+        other_moves = self.env['account.move'].search([('journal_id', '=', self.test_move.journal_id.id)]) - self.test_move
+        other_moves.unlink()  # Do not interfere when trying to get the highest name for new periods
+
+        moves = []
+        for date, name in [
+            ('2020-12-23', 'FA20209999'),
+            ('2020-12-30', '/'),
+        ]:
+            move = self.test_move.copy({'name': name, 'date': date})
+            if name != '/':
+                move.action_post()
+            moves.append(move)
+
+        move = moves[1]
+        move.action_post()
+        self.assertEqual(move.name, 'FA20210000')  # not what we expected
+
+        move.name = '/'
+        move.state = 'draft'
+        move.journal_id.sequence_override_regex = r'^(?P<prefix1>(?:R?FA)?)(?P<year>(?:\d{4})?)(?P<seq>\d*)(?P<suffix1>\D*?)$'  # set to custom regex
+        move.action_post()
+
+        self.assertEqual(move.name, 'FA202010000')  # good!
+        self.assertEqual(move.sequence_prefix, 'FA2020')
+        self.assertEqual(move.sequence_number, 10000)
+
+    def test_journal_override_sequence_no_reset_change_of_period(self):
+        """
+        Simple test to ensure that the no reset is taking place
+        """
+        other_moves = self.env['account.move'].search([('journal_id', '=', self.test_move.journal_id.id)]) - self.test_move
+        other_moves.unlink()  # Do not interfere when trying to get the highest name for new periods
+
+        moves = []
+        for date, name in [
+            ('2020-12-23', 'PREF/2020/3333'),
+            ('2021-01-02', '/'),
+        ]:
+            move = self.test_move.copy({'name': name, 'date': date})
+            if name != '/':
+                move.action_post()
+            moves.append(move)
+
+        move = moves[1]
+        move.journal_id.sequence_override_regex = r'^(?P<prefix1>(?:PREF/)?)(?P<year>(?:\d{4})?)(?P<prefix2>\D+?)(?P<seq>\d*)(?P<suffix>\D*?)$'  # set to custom regex
+        move.action_post()
+        self.assertEqual(move.name, 'PREF/2021/0001')  # ok, it was reset
+
+        move.journal_id.sequence_override_regex = r'^(?P<prefix1>(?:PREF/)?)(?P<year>(?:\d{4})?)(?P<prefix2>\D+?)(?P<seq>(?P<no_reset>)\d*)(?P<suffix>\D*?)$'  # set to not reset
+        move.name = '/'
+        move.state = 'draft'
+        move.action_post()
+        self.assertEqual(move.name, 'PREF/2021/3334')
+        self.assertEqual(move.sequence_prefix, 'PREF/2021/')
+        self.assertEqual(move.sequence_number, 3334)
+
+    def test_journal_override_sequence_bad_last_sequence(self):
```

Comment:
@william-andre This tests is to showcase the issue on current implementation (it passes). 
Ordering by ID [here](https://github.com/odoo/odoo/blob/c5225377cf733e169c7fa306d72c990ce493551b/addons/account/models/sequence_mixin.py#L185) seems to be the issue, when ordering by `date` we get the _expected_(?) result. What do you think?

This test does not relate to not resetting the sequence, so better we keep it out of the last version of this PR. Not sure if we should add it on another.

Reply:
It is the last prefix you used in that period, I don't think it is wrong.